File: clients/Scala/src/main/scala/model/AttackProperties.scala
package model

import util.StreamUtil

case class AttackProperties(attackRange: Int, damage: Int, collectResource: Boolean) {
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, attackRange)
        StreamUtil.writeInt(stream, damage)
        StreamUtil.writeBoolean(stream, collectResource)
    }
}
object AttackProperties {
    def readFrom(stream: java.io.InputStream): AttackProperties = AttackProperties(
        StreamUtil.readInt(stream)
        ,
        StreamUtil.readInt(stream)
        ,
        StreamUtil.readBoolean(stream)
        )
}

File: clients/JavaScript/model/attack-properties.js
class AttackProperties {
    constructor(attackRange, damage, collectResource) {
        this.attackRange = attackRange;
        this.damage = damage;
        this.collectResource = collectResource;
    }
    static async readFrom(stream) {
        let attackRange;
        attackRange = await stream.readInt();
        let damage;
        damage = await stream.readInt();
        let collectResource;
        collectResource = await stream.readBool();
        return new AttackProperties(attackRange, damage, collectResource);
    }
    async writeTo(stream) {
        let attackRange = this.attackRange;
        await stream.writeInt(attackRange);
        let damage = this.damage;
        await stream.writeInt(damage);
        let collectResource = this.collectResource;
        await stream.writeBool(collectResource);
    }
}
module.exports = AttackProperties
